test/channels.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const DC = require('diagnostics_channel');
+
+const Code = require('@hapi/code');
+const Lab = require('@hapi/lab');
+
+const Hapi = require('..');
+
+const { describe, it } = exports.lab = Lab.script();
+const expect = Code.expect;
+
+describe('DiagnosticChannel', () => {
+
+    describe('onServerChannel', () => {
+
+        const channel = DC.channel('hapi.onServer');
+
+        it('server should be exposed on creation through the channel hapi.onServer', async () => {
+
+            let exposedServer;
+            let server;
+
+            await new Promise((resolve) => {
+
+                channel.subscribe((srv) => {
+
+                    exposedServer = srv;
+                    resolve();
```

Comment:
It is simpler and more clear to resolve with the `server` and assign it the the awaited result of the promise.

```js
             const exposedServer = await new Promise((resolve) => {

                 channel.subscribe(resolve);
             });

             const server = Hapi.server();

             expect(exposedServer).to.shallow.equal(server);
```